### PR TITLE
Add all files required for user password reset

### PR DESCRIPTION
**Pull request by sebastiancaraballo**
Trello board reference:
https://trello.com/c/1FpW0xPg

Description:
Add all files required for user password reset.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -60,6 +59,7 @@ group :test do
   gem 'database_cleaner', '~> 1.4.1'
   gem 'shoulda-matchers', '~> 3.1.1'
   gem 'simplecov', '~> 0.13.0', require: false
+  gem 'addressable'
```
Comment (mdornel) [inline Gemfile]: Add gem version

Comment (nicofh) [inline Gemfile]: You should specify gem version

File: spec/requests/api/v1/passwords/create_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    let(:params) { { email: user.email } }
+
+    it 'returns a successful response' do
+      post user_password_path, params: params, as: :json
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'returns email sent notification' do
+      post user_password_path, params: params, as: :json
+      expect(json[:message]).to match(user.email)
```
Comment (mdornel) [inline spec/requests/api/v1/passwords/create_spec.rb:16]: Here I think that you are not expecting what the example is describing. Maybe you should expect the full notification or change the description.

File: spec/requests/api/v1/passwords/edit_spec.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  let(:params) do
+    {
+      reset_password_token: password_token,
+      redirect_url: ENV['PASSWORD_RESET_URL']
+    }
+  end
+
+  it 'returns access token, uid and client id for update request' do
+    get edit_user_password_path, params: params
+    expect(response.header['Location']).to include('token')
```
Comment (mdornel) [inline spec/requests/api/v1/passwords/edit_spec.rb]: Use `include_json` that you previously used.

File: Gemfile (hunk 1 above)
Comment (mdornel) [reply Gemfile]: By the way, I do not think that we need this gem

File: spec/requests/api/v1/passwords/update_spec.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
+require 'rails_helper'
+require 'addressable'
+
+describe 'PUT api/v1/users/passwords/', type: :request do
+  let(:user) { create(:user) }
+  let(:password_token) { user.send(:set_reset_password_token) }
+  let(:headers) do
```
Comment (mdornel) [inline spec/requests/api/v1/passwords/update_spec.rb]: I think you there is no need of this complete block

**Review by mdornel — CHANGES_REQUESTED** [review]

File: spec/requests/api/v1/passwords/update_spec.rb (hunk 4 above)
Comment (mdornel) [reply spec/requests/api/v1/passwords/update_spec.rb]: Sending the auth_header should be enough

**Review by mdornel — APPROVED** [review]

**Review by mdornel — COMMENTED** [review]
Remember to update the apiary

**Review by nicofh — APPROVED** [review]

**Review by mdornel — APPROVED** [review]

**Pull request merged**